src/StaticWebAssetsSdk/Tasks/DefineStaticWebAssets.Cache.cs

```diff
@@ -0,0 +1,217 @@
+        public HashSet<string> InputHashes { get; set; } = [];
+
+        // Outputs for the cache
+        public Dictionary<string, StaticWebAsset> CachedAssets { get; set; } = [];
+        public Dictionary<string, CopyCandidate> CachedCopyCandidates { get; set; } = [];
+
+        internal static DefineStaticWebAssetsCache ReadOrCreateCache(TaskLoggingHelper log, string manifestPath)
+        {
+            if (manifestPath != null && File.Exists(manifestPath))
+            {
+                using var existingManifestFile = File.OpenRead(manifestPath);
+                var cache = JsonSerializer.Deserialize(existingManifestFile, DefineStaticWebAssetsSerializerContext.Default.DefineStaticWebAssetsCache);
+                if (cache == null)
+                {
+                    throw new InvalidOperationException($"Failed to deserialize cache from {manifestPath}");
+                }
+                cache.SetPathAndLogger(manifestPath, log);
+                return cache;
+            }
+            else
+            {
+                return new DefineStaticWebAssetsCache(log, manifestPath);
+            }
+        }
+
+        internal void WriteCacheManifest()
+        {
+            if (_manifestPath != null)
+            {
+                using var manifestFile = File.OpenWrite(_manifestPath);
+                manifestFile.SetLength(0);
+                JsonSerializer.Serialize(manifestFile, this, DefineStaticWebAssetsSerializerContext.Default.DefineStaticWebAssetsCache);
+            }
+        }
+
+        internal void AppendAsset(string hash, StaticWebAsset asset, ITaskItem item)
+        {
+            asset.AssetKind = item.GetMetadata(nameof(StaticWebAsset.AssetKind));
+            _assets.Add(item);
+            CachedAssets[hash] = asset;
+        }
+
+        internal void AppendCopyCandidate(string hash, string identity, string targetPath)
+        {
+            var copyCandidate = new CopyCandidate(identity, targetPath);
+            _copyCandidates.Add(copyCandidate.ToTaskItem());
+            CachedCopyCandidates[hash] = copyCandidate;
+        }
+
+        internal void PrepareForProcessing(
+            byte[] propertiesHash,
+            byte[] fingerprintPatternsHash,
+            byte[] propertyOverridesHash,
+            Dictionary<string, ITaskItem> inputHashes)
+        {
+            if (!propertiesHash.SequenceEqual(GlobalPropertiesHash) ||
+                !fingerprintPatternsHash.SequenceEqual(FingerprintPatternsHash) ||
+                !propertyOverridesHash.SequenceEqual(PropertyOverridesHash))
+            {
+                TotalUpdate(propertiesHash, fingerprintPatternsHash, propertyOverridesHash, inputHashes);
+            }
+            else
+            {
+                PartialUpdate(inputHashes);
+            }
+        }
+
+        internal void TotalUpdate(byte[] propertiesHash, byte[] fingerprintPatternsHash, byte[] propertyOverridesHash, IDictionary<string, ITaskItem> inputsByHash)
```

Comment:
Nit: These methods being `internal` made me look for other places where they're called. As far as I can tell, these can be private. Should they?